### ISSUE TITLE
device-method CLI does not send proper JSON parameters

### ISSUE THREAD

**Issue by mwm-twx** [Azure/iothub-explorer#14]
Per the direct method documentation (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods), the recipient of a call will receive a JSON payload:
```javascript
{
    "input1": "someInput",
    "input2": "anotherInput"
}
```

Attempts to send a similar payload through the iothub-explorer have produced different results, but none are valid JSON.

`iothub-explorer device-method DEVICE METHOD {"foo": "bar"}` gives the receiver `{foo:bar}`.

`iothub-explorer device-method DEVICE METHOD '{"foo": "bar"}'` gives the receiver `"{\"foo\":\"bar\"}"`.  This is closer, but the leading and trailing quotes cause rejection from JSON parsers.

**Comment by pierreca** [comment]
What command line interpreter are you using?

**Comment by mwm-twx** [comment]
I'm using zsh.

**Comment by pierreca** [comment]
OK I installed zsh and tried the second method you mention and it worked fine (meaning I didn't get the escaped double quotes and the leading and trailing quotes didn't cause rejection from the parser). The client I was using is the Node.js SDK though.

Can you paste the code you're using to receive the payload, as well as the versions of iothub-explorer and of the client SDK that you are using?

**Comment by pierreca** [comment]
I'm seeing the editions to your comment just now, so now i'm not sure what you expect. is it to have request.payload to be an object instead of a string that you can JSON.parse()? 

Seeing the code you have or that you wish worked would help.

**Comment by mwm-twx** [comment]
I'm using the java device SDK to implement the method and expecting this to work:
```java
String textData = new String((byte[]) methodData, StandardCharsets.UTF_8);
JSONObject json = new JSONObject(textData);
```

The raw MQTT message received is:
```
000001c0  4b 22 7d 40 02 00 03 30  38 00 24 24 69 6f 74 68  |K"}@...08.$$ioth|
000001d0  75 62 2f 6d 65 74 68 6f  64 73 2f 50 4f 53 54 2f  |ub/methods/POST/|
000001e0  64 6f 53 74 75 66 66 2f  3f 24 72 69 64 3d 31 22  |doStuff/?$rid=1"|
000001f0  7b 5c 22 68 69 5c 22 3a  5c 22 79 6f 75 5c 22 7d  |{\"hi\":\"you\"}|
00000200  22 32 40 00 1f 24 69 6f  74 68 75 62 2f 6d 65 74  |"2@..$iothub/met|
```

The bytes received represent a quoted-string which happens to contain JSON, rather than containing a raw JSON string.

**Comment by mwm-twx** [comment]
Payload sent by iothub-explorer:
```javascript
{
  "methodName": "doStuff",
  "payload": "{\"hi\":\"you\"}",
  "timeoutInSeconds": 30
}
```

Expected payload (based on example in docs at https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods):
```javascript
{
  "methodName": "doStuff",
  "payload": {
    "hi": "you"
  },
  "timeoutInSeconds": 30
}
```

**Comment by pierreca** [comment]
I see - I guess iothub-explorer could try and parse the payload as JSON, and if it succeeds, set the payload in the request to the object instead of the JSON string.
I'll try and see if I can make that work, and in the meantime double-check with the Java SDK Maintainer to verify we're doing things by the book. 
Thanks again for taking the time to provide such complete feedback!

**Comment by pierreca** [comment]
@mwm-twx Here's a tentative fix (still has to pass code review)
https://github.com/Azure/iothub-explorer/tree/method-parse-payload

Would you mind trying this and confirm if it works as you expect?

**Comment by pierreca** [comment]
Just published version 1.1.11 that includes this improvement, closing the issue. Feel free to reopen if that's not the fix you're looking for!

**Comment by payalgaikwad42** [comment]
Hi,
i am getting this response on backend while trying to invoke direct method on my device. It seems like method is getting invoked but not working. can 
Response from onDev on edgedevice:
{
  "status": 400,
  "payload": "Invalid direct method parameter: This Is For Testing Purpose"
}
